backend/src/apiserver/server/pipeline_upload_server_test.go

```diff
@@ -534,22 +415,58 @@ func TestDefaultUpdatedPipelineVersion(t *testing.T) {
-	b = &bytes.Buffer{}
-	w = multipart.NewWriter(b)
-	part, _ = w.CreateFormFile("uploadfile", "arguments-version.tar.gz")
-	fileReader, _ = os.Open("test/arguments_tarball/arguments-version.tar.gz")
-	io.Copy(part, fileReader)
-	w.Close()
-	req, _ = http.NewRequest("POST", "/apis/v1beta1/pipelines/upload_version?pipelineid="+resource.DefaultFakeUUID, bytes.NewReader(b.Bytes()))
-	req.Header.Set("Content-Type", w.FormDataContentType())
-	rr = httptest.NewRecorder()
-	handler = http.HandlerFunc(server.UploadPipelineVersion)
-	handler.ServeHTTP(rr, req)
+	server = updateClientManager(clientManager, util.NewFakeUUIDGeneratorOrFatal(fakeVersionUUID, nil))
+	bytesBuffer, writer = setupWriter("")
+	setWriterFromFile("uploadfile", "arguments-version.tar.gz", "test/arguments_tarball/arguments-version.tar.gz", writer)
+	response = uploadPipeline(server, "/apis/v1beta1/pipelines/upload_version?pipelineid="+resource.DefaultFakeUUID,
+		bytes.NewReader(bytesBuffer.Bytes()), writer, server.UploadPipelineVersion)
+	assert.Equal(t, 200, response.Code)
 
 	pipeline, err := clientManager.PipelineStore().GetPipeline(resource.DefaultFakeUUID)
 	assert.Nil(t, err)
 	assert.Equal(t, pipeline.DefaultVersionId, fakeVersionUUID)
 }
+
+func setWriterWithBuffer(fieldname string, filename string, buffer string, writer *multipart.Writer) {
+	part, _ := writer.CreateFormFile(fieldname, filename)
+	io.Copy(part, bytes.NewBufferString(buffer))
+	writer.Close()
+}
+
+func setWriterFromFile(fieldname string, filename string, filepath string, writer *multipart.Writer) {
+	part, _ := writer.CreateFormFile(fieldname, filename)
+	fileReader, _ := os.Open(filepath)
+	io.Copy(part, fileReader)
+	writer.Close()
+}
+
+func setupWriter(text string) (*bytes.Buffer, *multipart.Writer) {
+	bytesBuffer := &bytes.Buffer{}
+	if text != "" {
+		bytesBuffer.WriteString(text)
+	}
+	return bytesBuffer, multipart.NewWriter(bytesBuffer)
+}
+
+func setupClientManagerAndServer() (*resource.FakeClientManager, PipelineUploadServer) {
+	clientManager := resource.NewFakeClientManagerOrFatal(util.NewFakeTimeForEpoch())
+	resourceManager := resource.NewResourceManager(clientManager)
+	server := PipelineUploadServer{resourceManager: resourceManager, options: &PipelineUploadServerOptions{CollectMetrics: false}}
+	return clientManager, server
+}
+
+func updateClientManager(clientManager *resource.FakeClientManager, uuid util.UUIDGeneratorInterface) PipelineUploadServer {
+	clientManager.UpdateUUID(uuid)
+	resourceManager := resource.NewResourceManager(clientManager)
+	server := PipelineUploadServer{resourceManager: resourceManager, options: &PipelineUploadServerOptions{CollectMetrics: false}}
+	return server
+}
+
+func uploadPipeline(server PipelineUploadServer, url string, body io.Reader, writer *multipart.Writer, uploadFunc func(http.ResponseWriter, *http.Request)) *httptest.ResponseRecorder {
```

Comment:
Sorry for the delay, I've been busy for a while.

Just realized one last thing, the `server` parameter doesn't seem to be used in the function body. Shall we remove it?

Reply:
Yeah, you are right. I should have double checked more carefully. Fixed.